### PR TITLE
prepend location for temporary module file to $MODULEPATH with high priority + mark it as default in load_fake_module method

### PR DESCRIPTION
**Pull request by boegel**
fix for #2499 reported by @mboisson based on suggestion by @ocaisa

### CODE REVIEW

File: easybuild/tools/modules.py [hunk 1]
```diff
@@ -330,8 +325,16 @@ def set_mod_paths(self, mod_paths=None):
...
-        """Add module path via 'module use'."""
+    def use(self, path, priority=None):
+        """
+        Add path to $MODULEPATH via 'module use'.
+
+        :param path: path to add to $MODULEPATH
+        :param priority: priority for this path in $MODULEPATH (Lmod-specific)
+        """
+        if priority:
+            self.log.info("Ignoring specified priority '%s' when prepending %s to $MODULEPATH", priority, path)
```
Comment (wpoely86) [inline easybuild/tools/modules.py]: I would add something here that it's called in `use`, not a `prepend_path`?

File: easybuild/tools/modules.py [hunk 2]
```diff
@@ -1212,17 +1208,33 @@ def update(self):
...
+        Add path to $MODULEPATH via 'module use'.
+
+        :param path: path to add to $MODULEPATH
+        :param priority: priority for this path in $MODULEPATH (Lmod-specific)
+        """
+        # make sure path exists before we add it
+        mkdir(path, parents=True)
+
+        if priority:
+            self.run_module(['use', '--priority', str(priority), path])
```
Comment (wpoely86) [inline easybuild/tools/modules.py]: I thought it was Lmod specific?

File: easybuild/tools/modules.py (hunk 1 above)
Comment (boegel) [reply easybuild/tools/modules.py]: fixed

File: easybuild/tools/modules.py (hunk 2 above)
Comment (boegel) [reply easybuild/tools/modules.py]: It is, this `use` implementation is custom to the `Lmod` class.

File: easybuild/tools/modules.py [hunk 3]
```diff
@@ -330,8 +325,16 @@ def set_mod_paths(self, mod_paths=None):
...
-        """Add module path via 'module use'."""
+    def use(self, path, priority=None):
+        """
+        Add path to $MODULEPATH via 'module use'.
+
+        :param path: path to add to $MODULEPATH
+        :param priority: priority for this path in $MODULEPATH (Lmod-specific)
+        """
+        if priority:
+            self.log.info("Ignoring specified priority '%s' when running 'module use %s'", priority, path)
```
Comment (migueldiascosta) [inline easybuild/tools/modules.py]: while in the code it's clear, I wonder if the log messages should also mention why the priority is ignored, for those not using `Lmod`

**Pull request merged**